examples/charm++/allGather/Makefile

```diff
@@ -0,0 +1,20 @@
+CHARMC = ../../../bin/charmc $(OPTS)
+FLAGS = -g -Wall -Wno-unused-variable -Wno-sign-compare -Wno-class-memaccess -O2
+TYPE ?= RING
+
+all: build
+
+build:
+	$(CHARMC) $(FLAGS) ./user.ci
+	$(CHARMC) $(FLAGS) -o allGather -D$(TYPE) ./user.cc -lmoduleallGather
+
+test: build
+	$(call run, +p19 ./allGather 19 29 7 )
```

Comment:
+p19 is too large for default testing. I think we usually keep it to +p6 or +p4 by default